// vite-plugin-csv/dist/index.mjs
// src/csv.ts
function loadCSV(code) {
  const csv = code.split(/\r\n|\n|\r/);
  const records = [];
  const headers = csv.slice(0)[0].split(",");
  const rows = csv.slice(1).filter((row) => row);
  for (let row of rows) {
    const record = row.split(",");
    if (record.every((item) => !item))
      continue;
    records.push(record);
  }
  return records.reduce((prev, next) => {
    const entry = {};
    headers.forEach((h, i) => {
      entry[h] = next[i];
    });
    prev.push(entry);
    return prev;
  }, []);
}

// src/index.ts
function VitePluginCSV() {
  return {
    name: "vite-plugin-csv",
    transform(code, id) {
      if (!id.includes(".csv"))
        return;
      try {
        const result = loadCSV(code);
        return {
          code: `export default ${JSON.stringify(result)};`,
          map: null
        };
      } catch (error) {
        this.error(error);
        return "";
      }
    }
  };
}
var src_default = VitePluginCSV;
export {
  src_default as default
};
